altsrc/flag.go

```diff
@@ -197,10 +197,8 @@ func (f *IntFlag) ApplyInputSourceValue(context *cli.Context, isc InputSourceCon
 			if err != nil {
 				return err
 			}
-			if value > 0 {
```

Comment:
It's hard to tell from the commit that introduced this check, because it happens to also be the commit that introduced YAML parsing, but I assume the intent here was to avoid calling `Set` on the flag set if the value wasn't explicitly specified. In an ideal world, I think we'd want `isc.Int` to return `value, ok, err`, so we would know whether the value was explicitly set as the zero value, versus just defaulting to zero. 

While it's also an imperfect solution, what are your thoughts on keeping a check for `value != 0` instead of `> 0`? We wouldn't know whether or not the value was actually set if it's zero, but the key difference in behavior is that we would report `false` for `IsSet`, which is closer to the current behavior, and, if I had to guess, more likely than the case where `0` is explicitly set.

Reply:
Actually thinking about it a bit more we want to check that the value from yaml is not the same as the default value for that flag and only then set it. Then this would take care of both the issues. i.e we would report IsSet only if the value from yaml is different from flag default and also allow for range of values. This way the value from yaml could be zero as well.

Reply:
@rliebz Looking at it further the isc.XXX(f.Name) will return an error if the flag isnt found in the input source. So there is no chance of the Set being called in that case. Added a small unit test to verify that

Reply:
FYI https://github.com/urfave/cli/issues/1373

I don't know if this is my code causing the problem, or the change in this PR, and I'm not knowledgeable enough to figure it out. Maybe you guys can find out?